channels/api.py

```diff
@@ -350,6 +354,63 @@ def sync_comment_model(*, channel_name, post_id, comment_id, parent_id):
         )[0]
 
 
+def sync_channel_subscription_model(channel_name, user):
+    """
+    Create or update channel subscription for a user
+
+    Args:
+        channel_name (str): The name of the channel
+        user (django.contrib.models.auth.User): The user
+
+    Returns:
+        ChannelSubscription: the channel user role object
+    """
+    with transaction.atomic():
+        channel = sync_channel_model(channel_name)
+        return ChannelSubscription.objects.update_or_create(channel=channel, user=user)[
+            0
+        ]
+
+
+def sync_channel_role_models(channel_name):
+    """
+    Get or create Channel roles
+
+    Args:
+        channel_name(str): The channel name
+
+    Returns:
+        list of ChannelGroupRole: the channel user role objects
+    """
+    with transaction.atomic():
+        channel = sync_channel_model(channel_name)
+        roles = {}
+        for role in [ROLE_MODERATORS, ROLE_CONTRIBUTORS]:
+            group, _ = Group.objects.get_or_create(
+                name="{}_{}".format(channel.name, role)
+            )
+            roles[role] = ChannelGroupRole.objects.get_or_create(
+                channel=channel, group=group, role=role
+            )[0]
+        return roles
+
+
+def update_user_role(channel_name, role, user, remove=False):
+    """
+    Add or remove a user from a channel role's group
+
+    Args:
+        channel_name(str): The channel name
+        user(django.contrib.auth.models.User): The user
+        role(str): The role name (moderators, contributors)
+        remove(bool): If the user should be removed instead of added
+    """
+    roles = sync_channel_role_models(channel_name)
```

Comment:
This will lead to 3 extraneous fetches from the database every time after the first when you just need the group. I think we need to move away from this sync_* pattern and instead ensure that these records are created when the channel is created and then in this situation simply fetch the group in one query. I've started doing this in my [PR for the article post](https://github.com/mitodl/open-discussions/pull/1452/files#diff-6b4e079385252444280613805ab05474R510), so that we can rely on that record being present if the channel is created. Since that all happens within the channel create api request, we can trust that we won't make any calls to add users before that is in place (if we do, then something is wrong).